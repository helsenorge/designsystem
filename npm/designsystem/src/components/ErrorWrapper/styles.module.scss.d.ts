export type Styles = {
  'field-set': string;
  'field-set__legend': string;
  'field-set__legend--on-dark': string;
  'form-group': string;
  'form-group--invalid': string;
  'form-group-wrapper': string;
  'form-group-wrapper__errors': string;
  'form-group-wrapper__title--on-dark': string;
};

export type ClassNames = keyof Styles;

declare const styles: Styles;

export default styles;
