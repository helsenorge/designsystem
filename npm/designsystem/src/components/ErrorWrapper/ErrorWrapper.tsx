import React from 'react';
import cn from 'classnames';

import styles from './styles.module.scss';

export interface ErrorWrapperProps {
  /** Error message */
  errorText?: string;
  /** Form component */
  children?: React.ReactNode;
}

export const ErrorWrapper: React.FC<ErrorWrapperProps> = props => {
  const { errorText } = props;

  const withErrorStyle = cn(styles['error-wrapper'], { [styles[`error-wrapper--with-error`]]: errorText });

  const errorStyles = cn(styles['error-wrapper__errors']);

  return (
    <div className={withErrorStyle}>
      {errorText && (
        <p role="alert" className={errorStyles}>
          {errorText}
        </p>
      )}
      {props.children}
    </div>
  );
};

export default ErrorWrapper;
