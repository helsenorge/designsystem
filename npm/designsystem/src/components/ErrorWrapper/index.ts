import ErrorWrapper from './ErrorWrapper';
export * from './ErrorWrapper';
export default ErrorWrapper;
