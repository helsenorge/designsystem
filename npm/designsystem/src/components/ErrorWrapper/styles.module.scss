@import '../../scss/_spacers.scss';
@import '../../scss/_breakpoints.scss';
@import '../../scss/_palette.scss';
@import '../../scss/font-settings.scss';

.error-wrapper {
  padding-bottom: getSpacer(m);
  @media (min-width: map-get($grid-breakpoints, md)) {
    padding-bottom: getSpacer(l);
  }

  &--with-error {
    padding-top: getSpacer(s);
    margin-left: getSpacer(xs) * -1;
    margin-right: getSpacer(xs) * -1;
    padding-left: getSpacer(2xs);
    padding-right: getSpacer(xs);
    background-color: $cherry100;
    border-left: getSpacer(3xs) solid $cherry500;

    @media (min-width: map-get($grid-breakpoints, md)) {
      margin-left: (getSpacer(m) + 0.25rem) * -1;
      margin-right: (getSpacer(m) + 0.25rem) * -1;
      padding-left: getSpacer(m);
      padding-right: getSpacer(m);
    }
  }

  &__errors {
    line-height: 1.875rem;
    font-size: $font-size-sm;
    font-weight: 600;
    color: $cherry500;
    margin-top: 0;
  }

  &__error-text {
    font-size: $font-size-sm;
    line-height: $font-size-sm;
    font-weight: 600;
    color: $cherry500;
    margin-top: 0;
  }
}
