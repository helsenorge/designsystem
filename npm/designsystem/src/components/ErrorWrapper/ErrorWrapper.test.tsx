import React from 'react';
import { render, screen } from '@testing-library/react';
import ErrorWrapper from './ErrorWrapper';

describe('Gitt at det skal vises en valideringsfeil', (): void => {
  describe('N책r ErrorWrapper rendres', (): void => {
    it('S책 vises Valideringsfeilen', (): void => {
      const { container } = render(<ErrorWrapper errorText="Dette er en feil" />);

      expect(container).toMatchSnapshot();
      const errorText = screen.getByRole('alert');
      expect(errorText).toBeVisible();
      expect(errorText.className).toBe('error-wrapper__errors');
      expect(errorText.textContent).toBe('Dette er en feil');

      expect(errorText.parentElement.className).toBe('error-wrapper error-wrapper--with-error');
    });
  });
});

describe('Gitt at det skal ikke vises en valideringsfeil', (): void => {
  describe('N책r ErrorWrapper rendres', (): void => {
    it('S책 vises ikke valideringsfeil', (): void => {
      const { container } = render(<ErrorWrapper />);

      expect(container).toMatchSnapshot();
    });
  });
});
