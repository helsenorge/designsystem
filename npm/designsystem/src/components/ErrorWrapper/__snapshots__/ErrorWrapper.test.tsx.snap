// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gitt at det skal ikke vises en valideringsfeil N책r ErrorWrapper rendres S책 vises ikke valideringsfeil 1`] = `
<div>
  <div
    class="error-wrapper"
  />
</div>
`;

exports[`Gitt at det skal vises en valideringsfeil N책r ErrorWrapper rendres S책 vises Valideringsfeilen 1`] = `
<div>
  <div
    class="error-wrapper error-wrapper--with-error"
  >
    <p
      class="error-wrapper__errors"
      role="alert"
    >
      Dette er en feil
    </p>
  </div>
</div>
`;
